utkdigitalinitiatives/TRACE
# Trac 1029e

** JIRA Ticket**: (link)
https://jira.lib.utk.edu/browse/TRAC-1029 [Create the Year-Index.htm page for trace.utk.edu]Other 

*Relevant Link
    https://scholar.google.com/intl/en/scholar/inclusion.html

#What does this pull request do?
Address TRAC-1029 by creating a world readable Search by year page for Google Scholar Inclusion.

The file to be revised is:  Trace/scripts/custom_scripts/create_pages.php.
a. add IBY_BODY code to build the HTML file presented for Google Scholar Inclusion
b. revise HOME_BODY code to create the link (http://localhost:8000/iby) to give access to the IBY_BODY page.

All of the searches described here search the field: mods_originInfo_encoding_edtf_keyDate_yes_dateIssued_dt
This field is for Publication Date.

Searches required by Google Scholar are Last-2-weeks, Search-by-year (going back at least 5 years).  

Other searches are included for use by our public audience.

*How should this be tested:

The only Collection that works for submission on vagrant is Graduate Theses and Dissertations.

    Checkout Trace(TRAC-1029e), do vagrant up.

    On the home page, click on the link: "Find latest publications! Search by date across all collections."
    -- This should take you to: http://localhost:8000/iby
    -- If not, contact me immediately. We have to deliver this for Google Scholar Inclusion.

    Add an etd by userA
    Make sure the Date of Award is in the ***PAST*** (my suggestion: date-of_award December 2016) or else the etd will not be seen in the collection until after the ***FUTURE*** date of award.
    SUBMIT. LOGOUT.

    LOGIN as thesis_manager, accept, publish.

    LOG OUT and become a non-authenticated user because these searches are supposed to be World-Readable.

    Click on the link for "Find latest..." ( localhost:8000/iby )
    Search on "Last 2 weeks".
    Even though dateModified for userA etd is very recent, this search should return nothing
    because date-of-award for userA etd is December-2016.
    If not, fail this test and fail the request.

    Go back to localhost:8000/iby
    Default test collections include date of publication 2016.
    Click on Search- by year: 2016
    This should return results, including the userA etd submission.
    If not, fail this test and fail the request.

    Go back to localhost:8000/iby
    Default test collections include date of publication 2008.
    Click on Search- by decade:  2000-2009
    This should return results.
    It should not return userA etd with date-of-award 2016.
    If not, fail this test and fail the request.

    If none of the three tests fail, pass the pull request.

Interested parties

@DonRichards
@CanOfBees
